arithmetization/src/main/java/net/consensys/linea/config/LineaProfitabilityConfiguration.java

```diff
@@ -16,17 +16,22 @@
 package net.consensys.linea.config;
 
 import lombok.Builder;
-import org.hyperledger.besu.datatypes.Wei;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.experimental.Accessors;
 
 /** The Linea profitability calculator configuration. */
 @Builder(toBuilder = true)
-public record LineaProfitabilityConfiguration(
-    int verificationGasCost,
-    int verificationCapacity,
-    int gasPriceRatio,
-    Wei gasPriceAdjustment,
-    double minMargin,
-    double estimateGasMinMargin,
-    double txPoolMinMargin,
-    boolean txPoolCheckApiEnabled,
-    boolean txPoolCheckP2pEnabled) {}
+@Accessors(fluent = true)
+@Getter
+@ToString
+public class LineaProfitabilityConfiguration {
+  @Setter private volatile long fixedCostKWei;
+  @Setter private volatile long variableCostKWei;
```

Comment:
This might be just good enough, but I had other idea. I think it would make more sense to formalize usage of these fields as follows:
Profitability logic for PENDING block (be it validation in block inclusion, or linea_estimateGas) should take extraData values from LATEST block's extra data. Meaning that if transaction was included into block #X, it means it was profitable according to #X-1's extraData.

It would make our inclusion logic more deterministic

Reply:
not sure I get your proposal correctly, because to me it seems it is already like that, since what happens now is:
1. a new block #X-1 is imported, extra data parsed and profitability conf updated
2. after some time a block #X creation starts, using profitability conf set by block #X-1

and 1 and 2 cannot overlap.

Reply:
If block #X-1 is imported at the same time as profitability formula being evaluated, isn't it possible for one of the 2 volatile values come from block X-2 and the other from X-1? Are we that confident about these values updates atomicity?

Reply:
And also, I'd expect all the blocks evaluated during import of X to be evaluated with the same configs from X-1, which is a stronger guarantee, not all individual transactions evaluated with the same configs

Reply:
good point about the atomicity, not enforced now, so I need to implement that.


> And also, I'd expect all the blocks evaluated during import of X to be evaluated with the same configs from X-1, which is a stronger guarantee, not all individual transactions evaluated with the same configs

_I'd expect all the blocks evaluated during import of X_ do you mean _txs_ instead of blocks?

when building a block, all the candidate pending txs are evaluated with the same profitability conf, in case a new block arrives in the meantime, the block building must restart anyway since the chain head changed.

Different is the case for `linea_estimateGas` and 'eth_sendRawTransaction` they could happen concurrently to block import

Reply:
> you mean txs instead of blocks?

Yes, sorry. All transactions